vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/AbstractConfigurationObject.java

```diff
@@ -161,6 +162,8 @@ protected void deepMarkAsDirty() {
      * resolved on the client-side.
      */
     protected void addChild(AbstractConfigurationObject configurationObject) {
+        Objects.requireNonNull(configurationObject,
+                "Child configuration object must not be null");
```

Comment:
Added an explicit check here. This is not a breaking change, the method would still have failed with a null pointer later on.

Also added a method that allows passing null, which is more convenient to use for objects that have nullable properties.